### DISCUSSION TITLE
Order of types in exported schema

### DISCUSSION THREAD

**Discussion by shodhansave** [strawberry-graphql/strawberry#3197] [General] [answered]
i have a route that returns the GralhQL schema (using [`print_schema`](/strawberry-graphql/strawberry/blob/main/strawberry/printer/printer.py#L553)). but the order in which the types are listed looks alphabetical. because of this, the `Query` type appears before the custom class' type. this looks sort of, confusing, because in the code we have defined the classes in the order of them being called (with `Query` being the last as it refers all the others). is it possible to generate the GraphQL schema based on the order in which the classes are defined?

**Comment by patrick91** [comment]
@shodhansave I'm not 100% we have a good way to do that if, the GraphQL schema generation starts when calling `strawberry.Schema` and then we traverse the schema from Query, Mutation and Subscription

What's the use case for having the same order?

**Reply by shodhansave** [reply]
@patrick91 there's no concrete use case as such. my use case is more about human-readability - usually the functions/classes in the code are defined in such order that the least called functions are on the top and the ones that call them are below. i was wondering if the schema can also look like that, with `Query` being in the end...

**Reply by patrick91** [reply]
we have had a similar request in the past: https://github.com/strawberry-graphql/strawberry/issues/3149

do you think that kind of order would work for you?

**Reply by shodhansave** [reply, marked as answer]
@patrick91 thanks for the link. im not sure this is what i wanted though, because it proposes of sorting the elements, whereas i was looking at the placements based on the placements of the corresponding python classes. nevermind, i do not have a concrete use case, so i would vote to close this discussion for now.

**Reply by patrick91** [reply]
ok! thanks for getting back to me 😊
